common/scala/src/main/scala/whisk/core/database/CouchDbRestStore.scala

```diff
@@ -169,6 +168,8 @@ class CouchDbRestStore[DocumentAbstraction <: DocumentSerializer](
                     transid.finished(this, start, s"[GET] '$dbName' failed to get document: '${doc}'; http status: '${code}'")
                     throw new Exception("Unexpected http response code: " + code)
             }
+        } recoverWith {
```

Comment:
don't we throw this exception from all the json serializers for the document types?
that's why i went for the top level error.